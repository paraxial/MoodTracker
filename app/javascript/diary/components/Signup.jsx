import React from 'react';
import TextInput from './TextInput';

class Signup extends React.Component {
  constructor() {
    super();
    this.state = {
      user: {
        email: null,
        password: null,
        confirmPassword: null,
      },
    };

    this.handleInputChange = this.handleInputChange.bind(this);
  }

  handleInputChange(field, value) {
    this.setState({ ...this.state, [field]: value });
  }

  render() {
    const { email, password, confirmPassword } = this.state.user;

    return (
        <React.Fragment>
          <h2 className="heading">
            Signup
          </h2>
          <div className="form-wrapper">
            <TextInput
              name="email"
              label="email"
              type="email"
              value={email}
              callback={this.handleInputChange}
            />
            <TextInput
              name="password"
              label="password"
              type="password"
              value={password}
              callback={this.handleInputChange}
            />
            <TextInput
              name="confirmPassword"
              label="confirm password"
              type="password"
              value={confirmPassword}
              callback={this.handleInputChange}
            />
            <input type="submit" className="action-button" value="Create Account"/>
          </div>
        </React.Fragment>
      );
  }
}

export default Signup;
