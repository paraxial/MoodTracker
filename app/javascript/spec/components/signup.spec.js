import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Signup from '../../diary/components/Signup';

describe('Signup', () => {
  test('should match snapshot', () => {
    const renderer = new ShallowRenderer();
    const page = renderer
      .render(<Signup />);

    expect(page).toMatchSnapshot();
  });
});
