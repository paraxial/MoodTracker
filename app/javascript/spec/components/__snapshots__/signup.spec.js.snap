// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup should match snapshot 1`] = `
<UNDEFINED>
  <h2
    className="heading"
  >
    Signup
  </h2>
  <div
    className="form-wrapper"
  >
    <TextInput
      callback={[Function]}
      label="email"
      name="email"
      type="email"
      value={null}
    />
    <TextInput
      callback={[Function]}
      label="password"
      name="password"
      type="password"
      value={null}
    />
    <TextInput
      callback={[Function]}
      label="confirm password"
      name="confirmPassword"
      type="password"
      value={null}
    />
    <input
      className="action-button"
      type="submit"
      value="Create Account"
    />
  </div>
</UNDEFINED>
`;
